ledger/complete/mtrie/node/node.go

```diff
@@ -122,12 +123,92 @@ func NewInterimNode(height int, lchild, rchild *Node) *Node {
 	return n
 }
 
+// deepCopy
+func (n *Node) deepCopy() *Node {
+	var h hash.Hash
+	copy(h[:], n.hashValue[:])
+
+	var p ledger.Path
+	copy(p[:], n.path[:])
+
+	var lChildCopy, rChildCopy *Node
+	if n.lChild != nil {
+		lChildCopy = n.lChild.deepCopy()
+	}
+	if n.rChild != nil {
+		rChildCopy = n.rChild.deepCopy()
+	}
+
+	return &Node{
+		lChild:    lChildCopy,
+		rChild:    rChildCopy,
+		height:    n.height,
+		path:      p,
+		payload:   n.payload.DeepCopy(),
+		maxDepth:  n.maxDepth,
+		regCount:  n.regCount,
+		hashValue: h,
+	}
+}
+
 // computeAndStoreHash computes the node's hash value and
 // stores the result in the nodes internal `hashValue` field
 func (n *Node) computeAndStoreHash() {
 	n.hashValue = n.computeHash()
 }
 
+// Pruned would look at the child of a node and if it can be pruned
+// it constructs a new subtrie and return that one instead of n
+// otherwise it would return the node itself if no change is required
+func (n *Node) Pruned() (*Node, bool) {
+	if n.IsLeaf() {
+		return n, false
+	}
```

Comment:
Leafs could also be unallocated registers that are explicitly contained in the trie. For unallocated subtries, we generally completely replace these with `nil`. I think we should do the same here:
```suggestion
	if n.IsADefaultNode() {
		return nil, n != nil
	}
	if n.IsLeaf() {
		return n, false
	}
```
Note that this requires the implementation of `IsADefaultNode()` to properly handle `nil` values, as suggested in my [earlier comment](https://github.com/onflow/flow-go/pull/1507#discussion_r741380958)